src/AryUtil.chpl

```diff
@@ -34,13 +34,52 @@ module AryUtil
       :arg name: name of the array
       :arg A: array to be printed
     */
-    proc formatAry(A):string throws {
-        if A.size <= printThresh {
-            return "%?".doFormat(A);
-        } else {
-            return "%? ... %?".doFormat(A[A.domain.low..A.domain.low+2],
-                                      A[A.domain.high-2..A.domain.high]);
+    proc formatAry(A: [?d]):string throws {
+        proc dimSummary(dimIdx: int): string throws {
+            const dimSize = d.dim(dimIdx).size;
+            var s: string;
+            if dimSize == 0 {
+                s = "";
+            } else if dimSize < printThresh {
+                var first = true,
+                    idx: d.rank*int;
+
+                for i in 0..<dimSize {
+                    if first then first = false; else s += " ";
+                    idx[dimIdx] = i;
+                    s += "?".doFormat(A[idx]);
+                }
+            } else {
+                var indices: 6*(d.rank*int);
+                indices[0][dimIdx] = 0;
+                indices[1][dimIdx] = 1;
+                indices[2][dimIdx] = 2;
+
+                for dd in 0..<d.rank {
+                    const dMax = d.dim(dd).high;
+                    indices[3][dd] = dMax;
+                    indices[4][dd] = dMax;
+                    indices[5][dd] = dMax;
+                }
+
+                indices[3][dimIdx] = dimSize-3;
+                indices[4][dimIdx] = dimSize-2;
+                indices[5][dimIdx] = dimSize-1;
```

Comment:
Yeah because of the zero based arrays those two queries end up being the same. Reading your comment and looking back through code makes me realize that it isn't super clear what this procedure is doing, so I'll comment it up a bit as well.